csrc/id_model/id_model.cpp

```diff
@@ -323,19 +325,42 @@ ValGraph& IdModel::buildExactGraph() {
           for (auto c_id :
                getSortedKeys(exact_c2p_root_map, Statement::lessThan)) {
             auto p_id = exact_c2p_root_map.at(c_id);
-            idGraph(IdMappingMode::EXACT).mapVals(c_id, p_id);
+            graph.mapVals(c_id, p_id);
           }
         }
       }
     }
 
     // TODO: Revisit if we really should map domains in the exact map
-    mapThroughLoopSwizzles(idGraph(IdMappingMode::EXACT));
+    mapThroughLoopSwizzles(graph);
+  }
+
+  // Map additional exact mappings if registered. Only map those that
+  // appear in this IdModel and when they are the same (per sameAs).
+  if (!tv_exprs_.empty()) {
+    Fusion* fusion = tv_exprs_.front()->fusion();
+    if (fusion->hasRegisteredExactMappings()) {
+      DisjointSets<IterDomain*> additional_mappings =
+          fusion->registeredExactMappings();
+      for (const auto& disjoint_set : additional_mappings.disjointSets()) {
+        auto num_ids = disjoint_set->size();
+        for (const auto i : c10::irange(num_ids)) {
+          for (const auto j : c10::irange(i + 1, num_ids)) {
```

Comment:
nit: Do I have to map all the pairs? Can I just `for (const auto i : c10::irange(1, num_ids))` and in the loop, just map `disjoint_set[i]` with `disjoint_set[0]` ?

Reply:
You're right. That should be sufficient. Will update.